jsonargparse/_link_arguments.py

```diff
@@ -81,7 +81,10 @@ def add_edge(self, source, target):
         for node in [source, target]:
             if node not in self.nodes:
                 self.nodes.append(node)
-        self.edges_dict[self.nodes.index(source)].append(self.nodes.index(target))
+        source_targets_list = self.edges_dict[self.nodes.index(source)]
+        target_index = self.nodes.index(target)
+        if target_index not in source_targets_list:
+            source_targets_list.append(target_index)
```

Comment:
Actually, this was not required.

Reply:
Noticed that it is needed. If there is a third link with the same parent, then an edge would be duplicated.